lib/dagDisplay.ml

```diff
@@ -854,7 +890,39 @@ let eval_predefined_apply f vl =
   | "max", s :: sl -> min_max_aux max s sl
   | _ -> raise Not_found
 
-let rec eval_var conf (page_title : Adef.safe_string)
+let parents_access_aux conf base td get_parent =
+  match td with
+  | TDitem (ip, _, _) | TDtext (ip, _) -> (
+      match get_parents (poi base ip) with
+      | Some ifam ->
+          let cpl = foi base ifam in
+          VVstring (Util.acces conf base (poi base (get_parent cpl)) :> string)
+      | None -> VVstring "")
+  | _ -> VVstring ""
+
+let has_sibling_aux base td next_or_prev =
+  match td with
+  | TDitem (ip, _, _) | TDtext (ip, _) -> (
+      match get_parents (poi base ip) with
+      | Some ifam ->
+          let sib = get_children (foi base ifam) in
+          (* array *)
+          let i = ref (-1) in
+          let _ = Array.iteri (fun n s -> if ip = s then i := n else ()) sib in
+          let cond =
+            if next_or_prev then !i >= 0 && !i < Array.length sib - 1
+            else !i >= 1
+          in
+          (cond, if cond then Some sib.(!i + if cond then 1 else -1) else None)
+      | None -> (false, None))
+  | _ -> (false, None)
```

Comment:
do you really need to return a tuple?
I think can just return an option

Reply:
Removed tuple.